ext/sockets/sockets.c

```diff
@@ -2046,7 +2044,19 @@ PHP_FUNCTION(socket_set_option)
 			optlen = sizeof(tv);
 			opt_ptr = &tv;
 #else
-			timeout = Z_LVAL_P(sec) * 1000 + Z_LVAL_P(usec) / 1000;
+			if (valsec < 0 || valsec > ULONG_MAX / 1000) {
+				zend_argument_value_error(4, "\"%s\" must be between 0 and %u", sec_key, (ULONG_MAX / 1000));
+				RETURN_THROWS();
+			}
+
+			timeout = valsec * 1000;
+
+			if (valusec < 0 || timeout > ULONG_MAX - (valusec / 1000)) {
```

Comment:
It seems to me that `ULONG_MAX - (valusec / 1000)` can yield a negative result, in which case we would throw. If we're fine with this (I think we are), the code is good as is.

Reply:
Maybe add a comment that `valuesec / 1000 > ULONG_MAX` is indeed considered a programmer error (i.e. throwing is deliberate here). Note that this code is only for Windows, where `ULONG_MAX` is 2^32, but `ZEND_LONG_MAX` is 2^63 for 64bit architectures.